src/infrasys/arrow_storage.py

```diff
@@ -99,51 +102,63 @@ def _(self, time_series):
         else:
             logger.debug("{} was already stored", time_series_uuid)
 
+    @_add_time_series.register(DeterministicTimeSeries)
+    def _(self, time_series):
+        """Store deterministic forecast time series data as a 2D matrix.
+
+        Each row represents a forecast window, and each column represents a time step
+        in the forecast horizon. The data is stored as a single array of arrays.
+        """
+        time_series_uuid = time_series.uuid
+        fpath = self._ts_directory.joinpath(f"{time_series_uuid}{EXTENSION}")
+
+        if not fpath.exists():
+            forecast_data = time_series.data_array
+
+            forecast_list = forecast_data.tolist()
+
+            schema = pa.schema([pa.field(str(time_series_uuid), pa.list_(pa.list_(pa.float64())))])
```

Comment:
I'm not sure I follow this. Why can't the schema be something like `[pa.field(str(time_series_uuid), col1, col2, ...]`